File: Python_HW/엄성민/hw_c++/perm.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
#include <string>
#include <cmath>
#include <sstream>

using namespace std;

vector<int> result2;
void pem(vector<int>& original, vector<int> result) {
    int size = original.size();
    
    if (size > 1) {
        for (int i = 0; i < size; i++) {
            vector<int> newVector = original;
            result.push_back(newVector[i]);
            newVector.erase(newVector.begin() + i);
            pem(newVector,result);
            result.pop_back();
        }
        
    }
    else if (size == 1) {
        result.push_back(original[0]);
        int val=0;
        for (int i = result.size() - 1; i >= 0; i--) {
            val += result[i] * pow(10, i);
        }
        auto it = find(result2.begin(), result2.end(), val);
        if (it == result2.end()) {
            result2.push_back(val);
        }
    }
}

int main() {

    string input;
    int count;
    vector<string> num1;
    vector<int> num2;
    cout << "숫자를 영어로 입력하세요" << endl;
    getline(cin, input);
    istringstream iss(input);
    string word;

    while (iss >> word) {
        num1.push_back(word);
    }

    cout << "숫자를 입력하세요"<<endl;
    cin >> count;
    string num_name[10] = {"zero", "one", "two", "three","four","five","six","seven","eight","nine" };
    int num[10] = {0,1,2,3,4,5,6,7,8,9 };
    
    for (int i = 0; i < num1.size(); i++) {
        for (int j = 0; j < 10; j++) {
            if (num1[i] == num_name[j]) {
                num2.push_back(num[j]);
            }
        }
    }
    vector<int> result;

    pem(num2, result);
    sort(result2.begin(), result2.end());
    count--;
    int final = result2[count];
    for (int i = num1.size() - 1; i >= 0; i--) {
        int pow_val = final / pow(10, i);
        for (int j = 0; j < 10; j++) {
            if (pow_val == num[j]) {
                cout << num_name[j] << " ";
            }
        }
        final -= pow_val * pow(10, i);
    }

    return 0;
}

File: Python_HW/서지민/perm.py
from itertools import combinations, permutations

sti = str(input())
num = int(input())
num = num -1

word = sti.split() #띄어쓰기 기준으로 나눠둠

for i in range(0, len(word)):
    if word[i] == 'three':
        word[i] = 3
    elif word[i] == 'two':
        word[i] = 2
    elif word[i] == 'one':
        word[i] = 1
    elif word[i] == 'zero':
        word[i] = 0   
word.sort()

perm = list(permutations(word, len(word)))
result = list(perm[num])

for i in range(0, len(result)):
    if result[i] == 3:
        result[i] = 'three'
    elif result[i] == 2:
        result[i] = 'two'
    elif result[i] == 1:
        result[i] = 'one'
    elif result[i] == 0:
        result[i] = 'zero'

str = " ".join(result)
print(str)